lib/faraday/utils.rb

```diff
@@ -95,7 +95,14 @@ def to_hash() ::Hash.new.update(self) end
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Find the last set of response headers.
+        start_index = headers.rindex { |x| x.match(/^HTTP\//) }
```

Comment:
Just for reference, it was a bug in VCR. Someone has already fixed it in a PR though 😄 https://github.com/vcr/vcr/pull/657